components/avatar-input.tsx

```diff
@@ -28,10 +28,34 @@ export default function AvatarFileInput(props: {
     [onChange],
   )
 
+  const handleEdit = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement | HTMLSpanElement>) => {
+      e.stopPropagation()
+      inputRef.current?.click()
+    },
+    [],
+  )
+
+  const handleDelete = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation()
+      onChange?.(undefined)
+    },
+    [onChange],
+  )
+
+  const handleClickAvatar = useCallback(
+    (e: React.MouseEvent<HTMLSpanElement>) => {
+      props.disabled ? undefined : handleEdit(e)
+    },
+    [handleEdit, props.disabled],
+  )
+
   return (
     <>
       <span
-        onClick={() => (props.disabled ? undefined : inputRef.current?.click())}
+        className="relative group block rounded-full"
+        onClick={handleClickAvatar}
```

Comment:
remove `handleClickAvatar` and use `props.disabled ? undefined : handleEdit` here

Reply:
Fixed.